### ISSUE TITLE
[BUG] darts relies on removed internal scikit-learn function _check_fit_params

### ISSUE THREAD

**Issue by rchui** [unit8co/darts#2174]
**Describe the bug**
Darts depends on a scikit-learn function `_check_fit_params` that was removed in scikit-learn==1.4.0

Traceback:
```python
/usr/local/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/usr/local/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
    exec(co, module.__dict__)
ml/tests/prediction/conftest.py:5: in <module>
    from ml.prediction.schema.tenant_forecast_properties import (
ml/prediction/schema/tenant_forecast_properties.py:10: in <module>
    from ml.prediction.forecasting.models import ModelManager
ml/prediction/forecasting/models.py:6: in <module>
    from darts.models import (
/usr/local/lib/python3.10/site-packages/darts/models/__init__.py:22: in <module>
    from darts.models.forecasting.linear_regression_model import LinearRegressionModel
/usr/local/lib/python3.10/site-packages/darts/models/forecasting/linear_regression_model.py:15: in <module>
    from darts.models.forecasting.regression_model import (
/usr/local/lib/python3.10/site-packages/darts/models/forecasting/regression_model.py:55: in <module>
    from darts.utils.multioutput import MultiOutputRegressor
/usr/local/lib/python3.10/site-packages/darts/utils/multioutput.py:5: in <module>
    from sklearn.utils.validation import _check_fit_params, has_fit_parameter
E   ImportError: cannot import name '_check_fit_params' from 'sklearn.utils.validation' (/usr/local/lib/python3.10/site-packages/sklearn/utils/validation.py)
```

**To Reproduce**
```
pip install darts==0.27.1 scikit-learn==1.4.0
```

Import `darts.models`

**Expected behavior**
I expect darts to not fail on import

**System (please complete the following information):**
 - Python version: 3.10
 - darts version >= 0.24.0

**Additional context**

**Comment by rchui** [comment]
An even more minimal example:

```python
docker run --rm -it python:3.10 bash

$ pip install darts
$ python
>>> import darts.models
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/darts/models/__init__.py", line 22, in <module>
    from darts.models.forecasting.linear_regression_model import LinearRegressionModel
  File "/usr/local/lib/python3.10/site-packages/darts/models/forecasting/linear_regression_model.py", line 15, in <module>
    from darts.models.forecasting.regression_model import (
  File "/usr/local/lib/python3.10/site-packages/darts/models/forecasting/regression_model.py", line 55, in <module>
    from darts.utils.multioutput import MultiOutputRegressor
  File "/usr/local/lib/python3.10/site-packages/darts/utils/multioutput.py", line 5, in <module>
    from sklearn.utils.validation import _check_fit_params, has_fit_parameter
ImportError: cannot import name '_check_fit_params' from 'sklearn.utils.validation' (/usr/local/lib/python3.10/site-packages/sklearn/utils/validation.py)
```

**Comment by dennisbader** [comment]
Hi @rchui, this is already fixed in the current master with #2170. Until it's released, you can install a lower sklearn version as described [here](https://github.com/unit8co/darts/pull/2170#issuecomment-1901073922).
